src/Microsoft.Extensions.Logging.AzureAppServices/Internal/BlobAppendReferenceWrapper.cs

```diff
@@ -1,46 +1,96 @@
-        /// <param name="cloudAppendBlob">The <see cref="CloudAppendBlob"/> instance to wrap.</param>
-        public BlobAppendReferenceWrapper(CloudAppendBlob cloudAppendBlob)
+        public BlobAppendReferenceWrapper(string containerUrl, string name, HttpClient client)
         {
-            _cloudAppendBlob = cloudAppendBlob;
+            var uriBuilder = new UriBuilder(containerUrl);
+            uriBuilder.Path += "/" + name;
+            _fullUri = uriBuilder.Uri;
+
+            AppendBlockQuery(uriBuilder);
+            _appendUri = uriBuilder.Uri;
+            _client = client;
         }
+
         /// <inheritdoc />
-        public async Task<Stream> OpenWriteAsync(CancellationToken cancellationToken)
+        public async Task AppendAsync(ArraySegment<byte> data, CancellationToken cancellationToken)
         {
-            return await _cloudAppendBlob.OpenWriteAsync(
-                createNew: false,
-                accessCondition: null,
-                options: null,
-                operationContext: null,
-                cancellationToken: cancellationToken);
+            Task<HttpResponseMessage> AppendDataAsync()
+            {
+                var message = new HttpRequestMessage(HttpMethod.Put, _appendUri)
+                {
+                    Content = new ByteArrayContent(data.Array, data.Offset, data.Count)
+                };
+                AddCommonHeaders(message);
+
+                return _client.SendAsync(message, cancellationToken);
+            }
+
+            var response = await AppendDataAsync();
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                // If no blob exists try creating it
+                var message = new HttpRequestMessage(HttpMethod.Put, _fullUri)
+                {
+                    // Set Content-Length to 0 to create "Append Blob"
+                    Content = new ByteArrayContent(Array.Empty<byte>()),
+                    Headers =
+                    {
+                        { "If-None-Match", "*" }
+                    }
+                };
+
+                AddCommonHeaders(message);
+
+                response = await _client.SendAsync(message, cancellationToken);
+
+                // If result is 2** or 412 try to append again
+                if (response.IsSuccessStatusCode ||
+                    response.StatusCode == HttpStatusCode.PreconditionFailed)
+                {
+                    // Retry sending data after blob creation
+                    response = await AppendDataAsync();
+                }
+            }
+
+            response.EnsureSuccessStatusCode();
         }
 
-        /// <inheritdoc />
-        public async Task CreateAsync(CancellationToken cancellationToken)
+        private static void AddCommonHeaders(HttpRequestMessage message)
+        {
+            message.Headers.Add("x-ms-blob-type", "AppendBlob");
+            message.Headers.Add("x-ms-version", "2015-12-11");
```

Comment:
:+1:
Based on this document support period is pretty long (version removed are 8 years old)
https://azure.microsoft.com/en-us/blog/microsoft-azure-storage-service-version-removal-update-extension-to-2016/